neqo-http3/src/priority.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{HFrame, Header};
+use std::fmt;
+use std::io::Write;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct Priority {
+    urgency: u8,
+    incremental: bool,
+}
+
+impl Default for Priority {
+    fn default() -> Self {
+        Priority {
+            urgency: 3,
+            incremental: false,
+        }
+    }
+}
+
+impl Priority {
+    /// # Panics
+    /// If an invalid urgency (>7 is given)
+    pub fn new(urgency: u8, incremental: bool) -> Priority {
+        assert!(urgency < 8);
+        Priority {
+            urgency,
+            incremental,
+        }
+    }
+
+    /// Returns a header if required to send
+    pub fn header(self) -> Option<Header> {
+        match self {
+            Priority {
+                urgency: 3,
+                incremental: false,
+            } => None,
+            Priority {
+                urgency: 3,
+                incremental: true,
+            } => Some(Header::new("priority", "i")),
+            Priority {
+                urgency,
+                incremental: false,
+            } => Some(Header::new("priority", format!("u={}", urgency))),
+            Priority {
+                urgency,
+                incremental: true,
+            } => Some(Header::new("priority", format!("u={},i", urgency))),
+        }
+    }
+
+    pub fn encode_request_frame(self, stream_id: u64) -> HFrame {
+        let mut priority = Vec::new();
+        write!(priority, "{}", self).unwrap();
+
+        HFrame::PriorityUpdateRequest {
+            element_id: stream_id,
+            priority,
+        }
+    }
+}
+
+impl fmt::Display for Priority {
```

Comment:
Is implementing Display a okay trait for this purpose?

Reply:
I think that this works nicely as it is.